src/cript/api/api.py

```diff
@@ -207,6 +208,13 @@ def _get_s3_client(self) -> boto3.client:  # type: ignore
 
         return s3_client
 
+    # Use a property to ensure delayed init of s3_client
+    @property
+    def _s3_client(self):
+        if self._internal_s3_client is None:
+            self._internal_s3_client = self._get_s3_client()
```

Comment:
I just noticed that the creation of the S3 client is split up along 2 different methods. I would recommend against this because it creates more confusion for the users and developers because they have to figure out which method does what when they are both essentially have the same purpose. I would just put all the code for the creation of the s3 client within a single method and be done with it

Reply:
I thought it was clearer, to keep the auth and contruction out of the caching mechanism.
But if you prefer it in a single function.
No problem. I combined them.